### PR TITLE
Drucker Prager yield stress prefactors

### PR DESCRIPTION
**Pull request by Geoniette**
Added yield stress prefactors to Drucker Prager material model. Test and test output also included.

### CODE REVIEW

File: include/aspect/material_model/rheology/drucker_prager.h [hunk 1]
```diff
@@ -107,6 +112,7 @@ namespace aspect
           double
           compute_yield_stress (const double cohesion,
                                 const double angle_internal_friction,
+                                const double yield_stress_prefactor,
```
Comment (naliboff) [inline include/aspect/material_model/rheology/drucker_prager.h]: I think there should actually be a way around adding this parameter here, in order to not break backwards compatibility. Hopefully the solution will be apparent later on.

File: include/aspect/material_model/rheology/drucker_prager.h [hunk 2]
```diff
@@ -137,6 +143,7 @@ namespace aspect
           double
           compute_viscosity (const double cohesion,
                              const double angle_internal_friction,
+                             const double yield_stress_prefactor,
```
Comment (naliboff) [inline include/aspect/material_model/rheology/drucker_prager.h]: Please remove this line

File: source/material_model/rheology/drucker_prager.cc [hunk 3]
```diff
@@ -128,6 +134,7 @@ namespace aspect
       double
       DruckerPrager<dim>::compute_viscosity (const double cohesion,
                                              const double angle_internal_friction,
+                                             const double yield_stress_prefactor,
```
Comment (naliboff) [inline source/material_model/rheology/drucker_prager.cc]: We will need to remove this

File: source/material_model/rheology/drucker_prager.cc [hunk 4]
```diff
@@ -136,6 +143,7 @@ namespace aspect
         DruckerPragerParameters p;
         p.angle_internal_friction = angle_internal_friction;
         p.cohesion = cohesion;
+        p.yield_stress_prefactor = yield_stress_prefactor;
```
Comment (naliboff) [inline source/material_model/rheology/drucker_prager.cc]: Remove this line

File: source/material_model/rheology/drucker_prager.cc [hunk 5]
```diff
@@ -80,12 +84,14 @@ namespace aspect
       double
       DruckerPrager<dim>::compute_yield_stress (const double cohesion,
                                                 const double angle_internal_friction,
+                                                const double yield_stress_prefactor,
```
Comment (naliboff) [inline source/material_model/rheology/drucker_prager.cc]: Remove this line

File: source/material_model/rheology/drucker_prager.cc [hunk 6]
```diff
@@ -80,12 +84,14 @@ namespace aspect
...
                                                 const double pressure,
                                                 const double max_yield_stress) const
       {
         DruckerPragerParameters p;
         p.angle_internal_friction = angle_internal_friction;
         p.cohesion = cohesion;
+        p.yield_stress_prefactor = yield_stress_prefactor;
```
Comment (naliboff) [inline source/material_model/rheology/drucker_prager.cc]: Remove this line

File: include/aspect/material_model/rheology/drucker_prager.h (hunk 1 above)
Comment (naliboff) [reply include/aspect/material_model/rheology/drucker_prager.h]: Yes, please remove this line

**Review by naliboff — COMMENTED** [review]
@Geoniette - Nice work getting this all sorted out and implemented!

I think we should proceed with the following steps:

1. Remove the modifications to the deprecated functions for compute_yield_stress and compute_viscosity (this is my fault, I got confused when we initially discussed the code).  I think it should compile.

2. Adjust the test output in the following manner:
- Rename `log.txt` to `screen-output`
- Remove all the current files related to visualization. If you would like to include visualization files as part of the test output, they should be output in gnu plot format and stay in the `solution/` folder.
- Remove original.prm, parameters.json, parameters.prm,

**Comment by naliboff** [comment]
@Geoniette @gassmoeller - a summary below of our discussion regarding the correct workflow steps for rebasing, in order to avoid the odd issues with commits showing up in the PR that had not been committed as part of the local branch:

```
1. git checkout main
2. git remote update
3. git pull upstream main
4. git checkout feature_branch_name
5. git rebase -i main
6. git push origin feature_branch_name
```

File: source/material_model/rheology/drucker_prager.cc [hunk 7]
```diff
@@ -116,12 +116,22 @@ namespace aspect
         // Here the middle circumscribing approach is taken.
         const double stress_inv_part = 1. / (std::sqrt(3.0) * (3.0 + sin_phi));
 
+        double yield_stress_prefactor;
+        if (std::isnan(p.yield_stress_prefactor))
```
Comment (naliboff) [inline source/material_model/rheology/drucker_prager.cc]: That was sneaky :)

File: source/material_model/rheology/drucker_prager.cc [hunk 8]
```diff
@@ -115,9 +119,9 @@ namespace aspect
         // Initial yield stress (no stabilization terms)
         const double yield_stress = ( (dim==3)
                                       ?
-                                      ( 6.0 * p.cohesion * cos_phi + 6.0 * pressure * sin_phi) * stress_inv_part
+                                      (( 6.0 * p.cohesion * cos_phi + 6.0 * pressure * sin_phi) * stress_inv_part) * p.angle_internal_friction
```
Comment (gassmoeller) [inline source/material_model/rheology/drucker_prager.cc]: doesnt this need to be `p.yield_stress_prefactor`? same for the line below

Comment (Geoniette) [reply source/material_model/rheology/drucker_prager.cc]: Yes! I had forgotten to add this back in after removing the if statement. Just fixed it now - thanks @gassmoeller

File: tests/drucker_prager_yield_stress_prefactors_0.prm [hunk 9]
```diff
@@ -0,0 +1,125 @@
+# This test checks that the yield prefactors stress is causing the
+# effective viscosity to change by the correct amount. The velocity
+# boundary conditions create a constant applied strain rate 3.1709e-16 1/s
+# (0.0005 m/y / 50e3 m / sec2yr). The drucker prager parameters create
+# a constant yield stress of 1e6 Pa, which would give a viscosity of
+# 1e6/2/3.17e-16 = 1.57e21 Pa s (this is the visco_plastic_yield test).
+# The maximum yield strength = 0.5e6 Pa, and so the effective
+# viscosity becomes 0.5e6/2/3.17e-16 = 7.88e20. Here we change the
+# yield stress prefactor to 0.5 which should give an effective viscosity
```
Comment (naliboff) [inline tests/drucker_prager_yield_stress_prefactors_0.prm:9]:
```suggestion
# yield stress prefactor to 0 which should give an effective viscosity
```

Comment (Geoniette) [reply tests/drucker_prager_yield_stress_prefactors_0.prm:9]: Thanks for catching this @naliboff! I've just fixed this in the last push along with the test fixes.

**Comment by naliboff** [comment]
@Geoniette - It looks like there are now quite a few modified by this PR. Is this expected with any recent changes?

**Review by gassmoeller — APPROVED** [review]
After some trials this should be ready to merge once the testers are done.
